<launch>
<!-- Launch file for ouster OS2-64 LiDAR -->
    <arg name="rviz" default="true" />

    <arg name="robot1_name" default="sct1"/>
    <arg name="robot2_name" default="sct2"/>


    <arg name="body_frame" default="ouster_lidar_link" />
    <arg name="init_frame" default="odom" />
    <arg name="base_frame" default="base_link" />

    <group ns="$(arg robot1_name)">
        <rosparam command="load" file="$(find simple_lio)/config/lio_ouster64.yaml" />

        <param name="lio_base/body_frame" type="string" value="$(arg robot1_name)/$(arg body_frame)"/>
        <param name="lio_base/init_frame" type="string" value="$(arg robot1_name)/$(arg init_frame)"/>
        <param name="lio_base/base_frame" type="string" value="$(arg robot1_name)/$(arg base_frame)"/>

        <node pkg="simple_lio" type="run_lio_mapping" name="run_lio_mapping" output="screen" />

        <group if="$(arg rviz)">
            <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz"
            args="-d $(find simple_lio)/config/rviz/simple_lio_$(arg robot1_name).rviz" />
        </group>
    </group>

    <group ns="$(arg robot2_name)">
        <rosparam command="load" file="$(find simple_lio)/config/lio_ouster64.yaml" />

        <param name="lio_base/body_frame" type="string" value="$(arg robot2_name)/$(arg body_frame)"/>
        <param name="lio_base/init_frame" type="string" value="$(arg robot2_name)/$(arg init_frame)"/>
        <param name="lio_base/base_frame" type="string" value="$(arg robot2_name)/$(arg base_frame)"/>

        <node pkg="simple_lio" type="run_lio_mapping" name="run_lio_mapping" output="screen" />

        <group if="$(arg rviz)">
            <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz"
            args="-d $(find simple_lio)/config/rviz/simple_lio_$(arg robot2_name).rviz" />
        </group>
    </group>

</launch>
